packages/remix-dev/compiler.ts

```diff
@@ -357,9 +357,15 @@ async function createBrowserBuild(
     mdxPlugin(config),
     browserRouteModulesPlugin(config, /\?browser$/),
     emptyModulesPlugin(config, /\.server(\.[jt]sx?)?$/),
-    // Must be placed before NodeModulesPolyfillPlugin, so yarn can resolve polyfills correctly
-    yarnPnpPlugin(),
     NodeModulesPolyfillPlugin(),
+    yarnPnpPlugin({
+      onLoad: async (args) => ({
+        contents: await fse.promises.readFile(args.path),
```

Comment:
`contents` and `loader` are the default values from the [default onLoad handler](https://github.com/yarnpkg/berry/blob/master/packages/esbuild-plugin-pnp/sources/index.ts#L53)